src/bjointsp/read_write/writer.py

```diff
@@ -90,9 +90,9 @@ def save_heuristic_variables(result, changed_instances, instances, edges, nodes,
             result["placement"]["flows"].append(flow)
         for path in e.paths:
             # record edge delay: all flows take the same (shortest) path => take path delay
-            delay = {"src": e.arc.source.name, "dest": e.arc.dest.name, "src_node": e.source.location, "dest_node": e.dest.location, "delay": sp.path_delay(links, path)}
```

Comment:
Could you please adjust this a little bit to have a list of `path_delays` that contains the original flow/link delays between interconnected VNFs and a separate block `vnf_delays` that contains the vnf processing delays. Mixing the two makes the results file harder to interpret.